src/service-override/files.ts

```diff
@@ -484,6 +498,15 @@ export default function getServiceOverride (): IEditorOverrideServices {
   }
 }
 
+/**
+ * Register a custom file system provider for the given schemes (the keys are the schemes,
+ * like "http", or "file"). This function needs to be called _before_ the file service is
+ * instantiated.
+ */
+export function registerCustomProviders (providers: CustomFileSystemProviders): void {
+  customFileSystemProviders = providers
+}
```

Comment:
What about being consistent with the `registerFileSystemOverlay` function by transforming it into `registerCustomProvider(scheme: string, provider)`

It could also simplify a bit the code:
```typescript
const providers = {
  user: new InMemoryFileSystemProvider(),
  extension: extensionFileSystemProvider,
  cache: new InMemoryFileSystemProvider(),
  logs: new InMemoryFileSystemProvider(),
  [Schemas.vscodeUserData]: new InMemoryFileSystemProvider(),
  [Schemas.tmp]: new InMemoryFileSystemProvider(),
  file: fileSystemProvider
}
export function registerCustomProvider (scheme: string, provider: transforming): void {
  customFileSystemProviders[scheme] = provider
}
```

Reply:
Good suggestion! I made the suggested changes.